packages/@o3r/schematics/src/modules/modules.display.rule.ts

```diff
@@ -1,15 +1,36 @@
 import type { Rule } from '@angular-devkit/schematics';
 import { EOL } from 'node:os';
+import { OTTER_MODULE_KEYWORD, OTTER_MODULE_SUPPORTED_SCOPES } from './modules.constants';
+
+/** Options for the Display Module rule */
+export interface DisplayModuleListOptions {
+  /**
+   * List of whitelisted scopes
+   * @default {@see OTTER_MODULE_SUPPORTED_SCOPES}
+   */
+  scopeWhitelist: string[] | readonly string[];
+  /**
+   * Keyword to search for Otter modules
+   * @default {@see OTTER_MODULE_KEYWORD}
+   */
+  keyword: string;
+  /** Display only the Otter modules with CMS administration */
+  onlyCmsModules: boolean;
+  /** Display only the Otter modules that are not already installed */
+  onlyNotInstalledModules: boolean;
+  /** Name of the package where to install add the dependency */
+  packageName: string;
+}
 
 /**
  * Display the list of available Otter modules
- *
- * @param keyword Keyword to search for Otter modules
- * @param scopeWhitelist List of whitelisted scopes
- * @param onlyCmsModules Display only the Otter modules with CMS administration
- * @param onlyNotInstalledModules Display only the Otter modules that are not already installed
+ * @param options Options of the Display Module list
  */
-export function displayModuleList(keyword: string, scopeWhitelist: string[] | readonly string[], onlyCmsModules = false, onlyNotInstalledModules = false): Rule {
+export function displayModuleListRule(options?: Partial<DisplayModuleListOptions>): Rule {
```

Comment:
a bit breaking. 
you could keep the old function as deprecated (not really mandatory)